boa_engine/src/vm/mod.rs

```diff
@@ -181,6 +181,16 @@ impl Vm {
 
         true
     }
+
+    pub(crate) fn get_return_value(&self) -> JsValue {
```

Comment:
Maybe rename `return_value` to  `completion_value`. That would align with the spec a bit more, right? Although, I'm sort of thinking this as long term it would be ideal to store the result/error value of the frame in `fp`, and that may not be the best way to approach it.

Reply:
> Maybe rename `return_value` to `completion_value`. That would align with the spec a bit 
more, right?

But the completion value can be an exception, when a [throw happens](https://tc39.es/ecma262/#sec-completion-record-specification-type) which `return_value` cannot be, so I think return_value is more appropriate.

> Although, I'm sort of thinking this as long term it would be ideal to store the result/error value of the frame in `fp`, and that may not be the best way to approach it.

In the latest commit I moved `return_value` to `Vm` instead of per function (call frame). 

This is simular to what v8 does with the `accumulator`, implicit return value. It may be a good idea to switch to a register-based VM at some point :)

Reply:
Sorry, I may have a bit unclear on the previous comment. I think the previous approach with the value on the stack works. I was more thinking it would be great to update `JsError`s conversion to and from `JsValue`, so instead of a `pending_exception` field and a `return_value` field, we have one field (a return value or completion value, however we want to name it) on the stack. We don't have a really good way currently to preserve `JsNativeError` currently though after converting it to an opaque, which was more so what I was thinking about when I had made the comment.

> It may be a good idea to switch to a register-based VM at some point :)

I was thinking the same thing :) But also see above